### PR TITLE
sync script

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: bin/sync.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+set -x
+
+if [ -z "$1" ]
+then
+    echo "Source IP provided, exiting"
```
Comment (elopio) [inline bin/sync.sh]: s/provided/not provided

File: bin/sync.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    exit 1
+fi
+if [ -z "$2" ]
+then
+    echo "Target IP provided, exiting"
```
Comment (elopio) [inline bin/sync.sh]: s/provided/not provided

File: bin/sync.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+sync_ssh_keys(){
+    dir=`mktemp -d`
+
+    scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ubuntu@$SOURCE_IP:$JENKINS_HOME/ssh-key/* $dir
+    scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no $dir/* ubuntu@$TARGET_IP:$JENKINS_HOME/ssh-key
+
+    for slave in vivid-1 xenial-1 xenial-2 xenial-3
+    do
+        ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ubuntu@$TARGET_IP \
+            sudo docker exec -t jenkins-slave-$slave bash -c "cp /var/jenkins_home/ssh-key/id_rsa* /home/jenkins-slave/.ssh"
```
Comment (elopio) [inline bin/sync.sh]: This probably needs more quoting. I'm getting:
cp: missing file operand
Try 'cp --help' for more information.

File: bin/sync.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+set -x
```
Comment (elopio) [inline bin/sync.sh]: I would use -e here, because the final rm hides the possible errors in the first steps.

File: bin/sync.sh (hunk 3 above)
Comment (fgimenez) [inline bin/sync.sh]: Thanks, fixed after escaping the quotes

File: bin/sync.sh (hunk 4 above)
Comment (fgimenez) [inline bin/sync.sh]: rsync complains about the status of some of the files, and with -e the whole process aborts. I've removed the rm, let me know what do you think

**Comment by fgimenez** [comment]
Thanks Leo, comments addressed, works for me (&trade;) :)

I've taken the opportunity of adding the sync of the jenkins credentials too, on a blank jenkins slate mo is happily connecting after a sync, pls give it a try.

File: bin/sync.sh (hunk 4 above)
Comment (elopio) [inline bin/sync.sh]: yes, it's tmp, it's alright.

**Comment by elopio** [comment]
:+1:

**Comment by fgimenez** [comment]
thx!

**Pull request merged**